cpu/lm4f120/periph/uart.c

```diff
@@ -31,120 +31,217 @@
  */
 static uart_isr_ctx_t config[UART_NUMOF];
 
-static int init_base(uart_t uart, uint32_t baudrate);
+typedef struct {
+    unsigned long uart_sysctl;         /**< UART device in sysctl */
+    unsigned long uart_base;           /**< UART base address */
+    IRQn_Type uart_irq_chan;           /**< UART IRQ channel */
+    volatile unsigned long * uart_im_r;/**< UART RX Enable Register */
+
+    unsigned long gpio_sysctl;         /**< GPIO device in sysctl */
+    unsigned long gpio_port;           /**< GPIO port */
+    unsigned long rx;              /**< pin used for RX */
+    unsigned long tx;              /**< pin used for TX */
+    unsigned long mask_rx;            /**< Pin mask for RX*/
+    unsigned long mask_tx;            /**< Pin mask for TX*/
+} uart_conf_fixed_t;
+
+static const uart_conf_fixed_t ucf[] = {
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART0,
+        .uart_base      = UART0_BASE,
+        .uart_irq_chan  = UART0_IRQn,
+        .uart_im_r      = &UART0_IM_R,
+
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOA,
+        .gpio_port   = GPIO_PORTA_BASE,
+        .rx             = GPIO_PA0_U0RX,
+        .tx             = GPIO_PA1_U0TX,
+        .mask_rx        = (GPIO_PIN_0),
+        .mask_tx        = (GPIO_PIN_1),
+    },
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART1,
+        .uart_base      = UART1_BASE,
+        .uart_irq_chan  = UART1_IRQn,
+        .uart_im_r      = &UART1_IM_R,
+
+        #ifndef UART1_USE_ALT_PIN
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOB,
+        .gpio_port   = GPIO_PORTB_BASE,
+        .rx             = GPIO_PB0_U1RX,
+        .tx             = GPIO_PB1_U1TX,
+        .mask_rx        = (GPIO_PIN_0),
+        .mask_tx        = (GPIO_PIN_1),
+        #else
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOC,
+        .gpio_port   = GPIO_PORTC_BASE,
+        .rx             = GPIO_PC4_U1RX,
+        .tx             = GPIO_PC5_U1TX,
+        .mask_rx        = (GPIO_PIN_4),
+        .mask_tx        = (GPIO_PIN_5),
+        #endif
+    },
+    {
+        .uart_sysctl    = SYSCTL_PERIPH_UART2,
+        .uart_base      = UART2_BASE,
+        .uart_irq_chan  = UART2_IRQn,
+        .uart_im_r      = &UART2_IM_R,
+
+        .gpio_sysctl = SYSCTL_PERIPH_GPIOD,
+        .gpio_port   = GPIO_PORTD_BASE,
+        .rx             = GPIO_PD6_U2RX,
+        .tx             = GPIO_PD7_U2TX,
+        .mask_rx        = (GPIO_PIN_6),
+        .mask_tx        = (GPIO_PIN_7),
+    },
+};
+
+#define VALID_UART_CONFIGS  (sizeof(ucf) / sizeof(ucf[0]))
+
+#define UART_ASSERT assert(uart < UART_NUMOF); \
+                    assert(UART_NUMOF <= VALID_UART_CONFIGS);\
+                    assert(uart_config[uart].uart_num < VALID_UART_CONFIGS);
 
 /**
  * Configuring the UART console
  */
 int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 {
     /* Check the arguments */
-    assert(uart == 0);
+    UART_ASSERT;
```

Comment:
The `uart_init` function should not assert its inputs, but check them and return fitting error codes -> as defined by the `periph/uart.h` API